File: src/com/codewars/kyu7/evenodddisparity/main/EvenOddDisparity.java
package com.codewars.kyu7.evenodddisparity.main;

public class EvenOddDisparity {
    public static int evenOddDisparity(int[] array) {
        int even = 0;
        int odd = 0;
        for (int i : array) {
            if (i % 2 == 0) {
                even++;
            } else {
                odd++;
            }
        }
        return even - odd;
    }
}

File: src/com/codewars/kyu7/evenodddisparity/python/EvenOddDisparity.py
def solve(a):
    even = []
    odd = []
    for i in a:
        if type(i) == int and i % 2 == 0:
            even.append(i)
        elif type(i) == int and i % 2 != 0:
            odd.append(i)
    return len(even) - len(odd)